File: shared-atlas-dart/lib/src/dto/signup_dto.dart
class SignupDto {
  final String username;
  final String password;
  final String invitationCode;
  final String? email;

  SignupDto({
    required this.username,
    required this.password,
    required this.invitationCode,
    this.email,
  });

  factory SignupDto.fromJson(Map<String, dynamic> json) {
    return SignupDto(
      username: json['username'] as String,
      password: json['password'] as String,
      invitationCode: json['invitationCode'] as String,
      email: json['email'] as String?,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'username': this.username,
      'password': this.password,
      'invitationCode': this.invitationCode,
      'email': this.email,
    };
  }
}

File: shared-atlas-rust/src/dto/signup_dto.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SignupDto {
    pub username: String,
    pub password: String,
    pub invitation_code: String,
    pub email: Option<String>,
}